catalog/transformer/catalog-transformer-zip/src/main/java/org/codice/ddf/catalog/transformer/zip/ZipCompression.java

```diff
@@ -0,0 +1,227 @@
+
+import ddf.catalog.CatalogFramework;
+import ddf.catalog.data.Attribute;
+import ddf.catalog.data.BinaryContent;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.Result;
+import ddf.catalog.data.impl.BinaryContentImpl;
+import ddf.catalog.data.impl.MetacardImpl;
+import ddf.catalog.operation.ResourceRequest;
+import ddf.catalog.operation.ResourceResponse;
+import ddf.catalog.operation.SourceResponse;
+import ddf.catalog.operation.impl.ResourceRequestByProductUri;
+import ddf.catalog.resource.Resource;
+import ddf.catalog.resource.ResourceNotFoundException;
+import ddf.catalog.resource.ResourceNotSupportedException;
+import ddf.catalog.transform.CatalogTransformerException;
+import ddf.catalog.transform.QueryResponseTransformer;
+
+import net.lingala.zip4j.core.ZipFile;
+import net.lingala.zip4j.exception.ZipException;
+import net.lingala.zip4j.model.ZipParameters;
+
+public class ZipCompression implements QueryResponseTransformer {
+
+    public CatalogFramework catalogFramework;
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ZipCompression.class);
+
+    /**
+     * Transforms a SourceResponse with a list of {@link Metacard}s into a {@link BinaryContent} item
+     * with an {@link InputStream}.  This transformation expects a key-value pair "fileName"-zipFileName to be present.
+     *
+     * @param upstreamResponse - a SourceResponse with a list of {@link Metacard}s to compress
+     * @param arguments        - a map of arguments to use for processing.  This method expects "fileName" to be set
+     * @return - a {@link BinaryContent} item with the {@link InputStream} for the Zip file
+     * @throws CatalogTransformerException when the transformation fails
+     */
+    public BinaryContent transform(SourceResponse upstreamResponse,
+            Map<String, Serializable> arguments) throws CatalogTransformerException {
+
+        if (upstreamResponse == null || CollectionUtils.isEmpty(upstreamResponse.getResults())) {
+            throw new CatalogTransformerException("No Metacards were found to transform.");
+        }
+
+        if (MapUtils.isEmpty(arguments) || !arguments.containsKey(ZipDecompression.FILE_PATH)) {
+            throw new CatalogTransformerException("No 'filePath' argument found in arguments map.");
+        }
+
+        ZipFile zipFile;
+        String filePath = (String) arguments.get(ZipDecompression.FILE_PATH);
+        try {
+            zipFile = new ZipFile(filePath);
+        } catch (ZipException e) {
+            LOGGER.debug("Unable to create zip file with path : {}", filePath, e);
+            throw new CatalogTransformerException(String.format("Unable to create zip file at %s",
+                    filePath), e);
+        }
+
+        List<Result> resultList = upstreamResponse.getResults();
+
+        Map<Resource, String> resourceMap = new HashMap<>();
+
+        resultList.forEach(result -> {
+            Metacard metacard = result.getMetacard();
```

Comment:
Since you only need the metacard in your `forEach`, you could add a `map(Result::getMetacard)` just before to make it a bit more explicit.